### PR TITLE
Added EpochLength to config

### LINKED ISSUE #1194: add BABE configuration option for epoch duration
## Expected Behavior

- for testing purposes, it would be useful to be able to configure epoch duration
- the config file should have a configuration option to manually set the BABE epoch duration (in slots)
- this should propagate to the `babe.ServiceConfig` 
- if `EpochDuration` is set in the babe config, it should overwrite the value read from the runtime
- set at genesis, won't be updated during network execution

## Current Behavior

- epoch duration must be configured via the runtime

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by RyRy79261**
## Changes

- Added EpochLength to config
- Updated the default config + toml's for Gossamer & Kusama
- Added a check in the create test service for BABE_test

## Tests

```
   go test ./lib/babe -run TestRunEpochLengthConfig -short
   go test ./... - short
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1194

### CODE REVIEW

File: chain/gssmr/config.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ roles = 4
 babe-authority = true
 grandpa-authority = true
 babe-threshold = ""
+epoch-length = 10
```
Comment (noot) [inline chain/gssmr/config.toml:28]: you can leave this out of this file, this should only be used for development

File: chain/gssmr/defaults.go [hunk 2]
```diff
@@ -54,6 +54,8 @@ var (
 	DefaultAuthority = true
 	// DefaultRoles Default node roles
 	DefaultRoles = byte(4) // authority node (see Table D.2)
+	// DefaultEpochLength is number of slots in an Epoch
+	DefaultEpochLength = uint64(10)
```
Comment (noot) [inline chain/gssmr/defaults.go:58]: can you change to 200?

File: chain/ksmcc/defaults.go [hunk 3]
```diff
@@ -52,6 +52,8 @@ var (
 
 	// DefaultAuthority true if BABE block producer
 	DefaultAuthority = false
+	// DefaultEpochLength is number of slots in an Epoch
+	DefaultEpochLength = uint64(10)
```
Comment (noot) [inline chain/ksmcc/defaults.go:56]: same here

**Review by noot — COMMENTED** [review]
can you also update `dot/services.go` `createBABEService` to update the `babe.ServiceConfig` also?

File: chain/gssmr/defaults.go (hunk 2 above)
Comment (RyRy79261) [reply chain/gssmr/defaults.go:58]: Sure thing

File: chain/gssmr/defaults.go [hunk 4]
```diff
@@ -39,7 +39,7 @@ var (
 	// InitConfig
 
 	// DefaultGenesisRaw Default genesis configuration path
-	DefaultGenesisRaw = string("./chain/gssmr/genesis-raw.json")
+	DefaultGenesisRaw = string("../../chain/gssmr/genesis-raw.json")
```
Comment (noot) [inline chain/gssmr/defaults.go:42]: can you change these back? now getting:
```
$ ./bin/gossamer init --force
EROR[12-09|10:16:21] failed to load genesis from file         pkg=cmd error="open /home/elizabeth/go/src/github.com/chain/gssmr/genesis-raw.json: no such file or directory" caller=config.go:630
INFO[12-09|10:16:21] reinitializing node...                   pkg=cmd basepath=/home/elizabeth/.gossamer/gssmr caller=main.go:280
INFO[12-09|10:16:21] initializing node...                     pkg=dot name=gssmr id=gssmr basepath=/home/elizabeth/.gossamer/gssmr genesis-raw=../../chain/gssmr/genesis-raw.json caller=node.go:53
EROR[12-09|10:16:21] failed to initialize node                pkg=cmd error="failed to load genesis from file: open /home/elizabeth/go/src/github.com/chain/gssmr/genesis-raw.json: no such file or directory" caller=main.go:296
failed to load genesis from file: open /home/elizabeth/go/src/github.com/chain/gssmr/genesis-raw.json: no such file or directory
```

File: chain/gssmr/defaults.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ var (
 	// DefaultID Default chain ID
 	DefaultID = string("gssmr")
 	// DefaultConfig Default toml configuration path
-	DefaultConfig = string("./chain/gssmr/config.toml")
+	DefaultConfig = string("../../chain/gssmr/config.toml")
```
Comment (noot) [inline chain/gssmr/defaults.go:32]: change back pls

File: chain/ksmcc/config.toml [hunk 6]
```diff
@@ -23,6 +23,7 @@ unlock = ""
 [core]
 roles = 1
 babe-authority = false
+epoch-length = 10
```
Comment (noot) [inline chain/ksmcc/config.toml:26]: oh pls remove from here also

File: cmd/gossamer/config_test.go [hunk 7]
```diff
@@ -279,6 +279,7 @@ func TestCoreConfigFromFlags(t *testing.T) {
 				Roles:            4,
 				BabeAuthority:    true,
 				GrandpaAuthority: true,
+				EpochLength:      10,
```
Comment (noot) [inline cmd/gossamer/config_test.go:282]: might need to update this since default epoch length changed

File: cmd/gossamer/config_test.go [hunk 8]
```diff
@@ -290,6 +291,7 @@ func TestCoreConfigFromFlags(t *testing.T) {
 				Roles:            0,
 				BabeAuthority:    false,
 				GrandpaAuthority: false,
+				EpochLength:      10,
```
Comment (noot) [inline cmd/gossamer/config_test.go:294]: same here

**Review by noot — APPROVED** [review]
looks good :D

**Review by arijitAD — APPROVED** [review]
`TestCoreConfigFromFlags` seems to be failing due to the epoch length mismatch. Otherwise looks good!

**Review by edwardmack — APPROVED** [review]
Nice job, lgtm.

**Pull request merged**